Shared/Samples/Search for web map/SearchForWebMapView.Model.swift

```diff
@@ -0,0 +1,121 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension SearchForWebMapView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A portal to ArcGIS Online to get the portal items from.
+        private let portal = Portal.arcGISOnline(connection: .anonymous)
+        
+        /// The query parameters for the next set of results based on the last results.
+        private var nextQueryParameters: PortalQueryParameters?
+        
+        /// The task used to find portal times thorough the portal.
+        private var task: Task<Void, Error>?
```

Comment:
I still tend to use https://developer.apple.com/documentation/swiftui/view/task(id:priority:_:) instead of a self managed task here. See comments below.

Reply:
I didn't go with the `.task` modifiers for a couple of reasons:

1. There should probably only be one search task going at a time. If there were two task modifiers and the user were to start loading the next results and then quickly change the query, the next results would load with the new results from the new search.
2. Task modifiers run when the view appears, so if there was one watching the query, it would reload the results unnecessarily when the user entered and exited a map even if the query didn't change, causing the user to lose their place in the results list.
3. IMO, this solution is more readable and less complex as it doesn't require extra state variables to get the task modifiers to play nice.

Thoughts?

Reply:
Well, I think only having 1 task makes sense. If you go this route, please make sure the errors from each Task block are thrown, or handled within the model. Currently the errors are "swallowed" by the Task blocks.

Reply:
I'd suggest either trying out the task modifier route, or handle the error within the model. The problem with managing Task block within the model is that, it basically turns methods that should be `async` into synchronous, and we don't have control over their lifespans.

You may try sth like below that handles the result of the task each time it is changed, but this IMHO is pretty close to just calling the async methods in task modifiers 😹 

```swift
.task(id: model.task) {
    guard let task = model.task else { return }
    let result = await task.result
    switch result {
    case .success(let success):
        <#code#>
    case .failure(let failure):
        <#code#>
    }
}
```

Or you may use a `do {} catch {}` block with every throwing method, and store a published error on the model.